src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -799,6 +800,12 @@ private void Parse()
             {
                 throw new MongoConfigurationException("This is an invalid w and journal pair.");
             }
+
+            string protectConnectionString(string connectionString)
+            {
+                var regex = new Regex("://(?<credentials>.*)@");
+                return regex.ReplaceGroup(connectionString, "credentials", "<hidden>");
```

Comment:
@jyemin will it be enough to hide all content between: `://` and `@`?

Reply:
Yes, that's enough to hide.  What does this do if there are no credentials.  No-op?

Reply:
>if there are no credentials. No-op?

yep